File: src/theme/colors.ts
export const lightColors = {
  // Primary colors
  primary: {
    main: '#6366F1',      // Indigo
    light: '#818CF8',
    dark: '#4F46E5',
    contrast: '#FFFFFF',
  },

  // Secondary colors
  secondary: {
    main: '#8B5CF6',      // Purple
    light: '#A78BFA',
    dark: '#7C3AED',
    contrast: '#FFFFFF',
  },

  // Semantic colors
  success: {
    main: '#10B981',      // Green
    light: '#34D399',
    dark: '#059669',
    background: '#D1FAE5',
  },

  error: {
    main: '#EF4444',      // Red
    light: '#F87171',
    dark: '#DC2626',
    background: '#FEE2E2',
  },

  warning: {
    main: '#F59E0B',      // Amber
    light: '#FBBF24',
    dark: '#D97706',
    background: '#FEF3C7',
  },

  info: {
    main: '#3B82F6',      // Blue
    light: '#60A5FA',
    dark: '#2563EB',
    background: '#DBEAFE',
  },

  // Neutral colors
  neutral: {
    50: '#F9FAFB',
    100: '#F3F4F6',
    200: '#E5E7EB',
    300: '#D1D5DB',
    400: '#9CA3AF',
    500: '#6B7280',
    600: '#4B5563',
    700: '#374151',
    800: '#1F2937',
    900: '#111827',
  },

  // Background colors
  background: {
    primary: '#FFFFFF',
    secondary: '#F9FAFB',
    tertiary: '#F3F4F6',
    dark: '#111827',
  },

  // Text colors
  text: {
    primary: '#111827',
    secondary: '#6B7280',
    tertiary: '#9CA3AF',
    disabled: '#D1D5DB',
    inverse: '#FFFFFF',
  },

  // Border colors
  border: {
    light: '#E5E7EB',
    main: '#D1D5DB',
    dark: '#9CA3AF',
  },

  // Overlay colors
  overlay: {
    light: 'rgba(0, 0, 0, 0.1)',
    medium: 'rgba(0, 0, 0, 0.3)',
    dark: 'rgba(0, 0, 0, 0.6)',
  },
};

export const darkColors = {
  // Primary colors
  primary: {
    main: '#818CF8',      // Lighter Indigo for dark mode
    light: '#A5B4FC',
    dark: '#6366F1',
    contrast: '#FFFFFF',
  },

  // Secondary colors
  secondary: {
    main: '#A78BFA',      // Lighter Purple for dark mode
    light: '#C4B5FD',
    dark: '#8B5CF6',
    contrast: '#FFFFFF',
  },

  // Semantic colors
  success: {
    main: '#34D399',      // Lighter Green for dark mode
    light: '#6EE7B7',
    dark: '#10B981',
    background: '#064E3B',
  },

  error: {
    main: '#F87171',      // Lighter Red for dark mode
    light: '#FCA5A5',
    dark: '#EF4444',
    background: '#7F1D1D',
  },

  warning: {
    main: '#FBBF24',      // Lighter Amber for dark mode
    light: '#FCD34D',
    dark: '#F59E0B',
    background: '#78350F',
  },

  info: {
    main: '#60A5FA',      // Lighter Blue for dark mode
    light: '#93C5FD',
    dark: '#3B82F6',
    background: '#1E3A8A',
  },

  // Neutral colors (reversed for dark mode)
  neutral: {
    50: '#111827',
    100: '#1F2937',
    200: '#374151',
    300: '#4B5563',
    400: '#6B7280',
    500: '#9CA3AF',
    600: '#D1D5DB',
    700: '#E5E7EB',
    800: '#F3F4F6',
    900: '#F9FAFB',
  },

  // Background colors
  background: {
    primary: '#000000',    // Pure black
    secondary: '#0A0A0A',  // Slightly lighter black
    tertiary: '#1A1A1A',   // Dark gray
    dark: '#000000',       // Pure black
  },

  // Text colors
  text: {
    primary: '#FFFFFF',    // Pure white
    secondary: '#E5E5E5',  // Light gray
    tertiary: '#B3B3B3',   // Medium gray
    disabled: '#6B7280',
    inverse: '#000000',    // Black
  },

  // Border colors
  border: {
    light: '#1A1A1A',
    main: '#2A2A2A',
    dark: '#3A3A3A',
  },

  // Overlay colors
  overlay: {
    light: 'rgba(0, 0, 0, 0.3)',
    medium: 'rgba(0, 0, 0, 0.5)',
    dark: 'rgba(0, 0, 0, 0.8)',
  },
};

export const colors = lightColors;
export type Colors = typeof lightColors;

File: GreatFeelSwiftUI/GreatFeelSwiftUI/Theme/Colors.swift
//
//  Colors.swift
//  GreatFeelSwiftUI
//
//  Design system colors matching the original React Native theme
//

import SwiftUI

struct AppColors {
    // MARK: - Light Theme
    struct Light {
        // Primary Colors
        static let primary = Color(hex: "6366F1")        // Indigo
        static let secondary = Color(hex: "8B5CF6")      // Purple
        static let accent = Color(hex: "F59E0B")         // Amber

        // Semantic Colors
        static let success = Color(hex: "10B981")        // Green
        static let error = Color(hex: "EF4444")          // Red
        static let warning = Color(hex: "F59E0B")        // Amber
        static let info = Color(hex: "3B82F6")           // Blue

        // Background Colors
        static let background = Color(hex: "FFFFFF")
        static let backgroundSecondary = Color(hex: "F9FAFB")
        static let backgroundTertiary = Color(hex: "F3F4F6")

        // Surface Colors
        static let surface = Color(hex: "FFFFFF")
        static let surfaceSecondary = Color(hex: "F9FAFB")

        // Text Colors
        static let textPrimary = Color(hex: "111827")
        static let textSecondary = Color(hex: "6B7280")
        static let textTertiary = Color(hex: "9CA3AF")
        static let textDisabled = Color(hex: "D1D5DB")

        // Border Colors
        static let border = Color(hex: "E5E7EB")
        static let borderFocus = Color(hex: "6366F1")

        // Neutral Scale
        static let neutral50 = Color(hex: "F9FAFB")
        static let neutral100 = Color(hex: "F3F4F6")
        static let neutral200 = Color(hex: "E5E7EB")
        static let neutral300 = Color(hex: "D1D5DB")
        static let neutral400 = Color(hex: "9CA3AF")
        static let neutral500 = Color(hex: "6B7280")
        static let neutral600 = Color(hex: "4B5563")
        static let neutral700 = Color(hex: "374151")
        static let neutral800 = Color(hex: "1F2937")
        static let neutral900 = Color(hex: "111827")
    }

    // MARK: - Dark Theme
    struct Dark {
        // Primary Colors
        static let primary = Color(hex: "818CF8")        // Lighter Indigo
        static let secondary = Color(hex: "A78BFA")      // Lighter Purple
        static let accent = Color(hex: "FBBf24")         // Lighter Amber

        // Semantic Colors
        static let success = Color(hex: "34D399")        // Lighter Green
        static let error = Color(hex: "F87171")          // Lighter Red
        static let warning = Color(hex: "FBBF24")        // Lighter Amber
        static let info = Color(hex: "60A5FA")           // Lighter Blue

        // Background Colors
        static let background = Color(hex: "000000")
        static let backgroundSecondary = Color(hex: "0A0A0A")
        static let backgroundTertiary = Color(hex: "141414")

        // Surface Colors
        static let surface = Color(hex: "1A1A1A")
        static let surfaceSecondary = Color(hex: "262626")

        // Text Colors
        static let textPrimary = Color(hex: "F9FAFB")
        static let textSecondary = Color(hex: "D1D5DB")
        static let textTertiary = Color(hex: "9CA3AF")
        static let textDisabled = Color(hex: "4B5563")

        // Border Colors
        static let border = Color(hex: "374151")
        static let borderFocus = Color(hex: "818CF8")

        // Neutral Scale (reversed)
        static let neutral50 = Color(hex: "111827")
        static let neutral100 = Color(hex: "1F2937")
        static let neutral200 = Color(hex: "374151")
        static let neutral300 = Color(hex: "4B5563")
        static let neutral400 = Color(hex: "6B7280")
        static let neutral500 = Color(hex: "9CA3AF")
        static let neutral600 = Color(hex: "D1D5DB")
        static let neutral700 = Color(hex: "E5E7EB")
        static let neutral800 = Color(hex: "F3F4F6")
        static let neutral900 = Color(hex: "F9FAFB")
    }

    // MARK: - Category Colors (for goals and content types)
    struct Category {
        static let breath = Color(hex: "3B82F6")         // Blue
        static let meditation = Color(hex: "8B5CF6")     // Purple
        static let sleep = Color(hex: "6366F1")          // Indigo
        static let relax = Color(hex: "10B981")          // Green
        static let mindfulness = Color(hex: "F59E0B")    // Amber
        static let stress = Color(hex: "EF4444")         // Red
        static let anxiety = Color(hex: "EC4899")        // Pink
        static let productivity = Color(hex: "14B8A6")   // Teal
    }
}

// MARK: - Color Extension for Hex Support
extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3: // RGB (12-bit)
            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6: // RGB (24-bit)
            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8: // ARGB (32-bit)
            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default:
            (a, r, g, b) = (255, 0, 0, 0)
        }

        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue:  Double(b) / 255,
            opacity: Double(a) / 255
        )
    }
}

// MARK: - Theme-aware Color Provider
extension AppColors {
    static func primary(for colorScheme: ColorScheme) -> Color {
        colorScheme == .dark ? Dark.primary : Light.primary
    }

    static func secondary(for colorScheme: ColorScheme) -> Color {
        colorScheme == .dark ? Dark.secondary : Light.secondary
    }

    static func background(for colorScheme: ColorScheme) -> Color {
        colorScheme == .dark ? Dark.background : Light.background
    }

    static func textPrimary(for colorScheme: ColorScheme) -> Color {
        colorScheme == .dark ? Dark.textPrimary : Light.textPrimary
    }

    static func textSecondary(for colorScheme: ColorScheme) -> Color {
        colorScheme == .dark ? Dark.textSecondary : Light.textSecondary
    }

    static func surface(for colorScheme: ColorScheme) -> Color {
        colorScheme == .dark ? Dark.surface : Light.surface
    }

    static func border(for colorScheme: ColorScheme) -> Color {
        colorScheme == .dark ? Dark.border : Light.border
    }

    static func textTertiary(for colorScheme: ColorScheme) -> Color {
        colorScheme == .dark ? Dark.textTertiary : Light.textTertiary
    }
}
